code/game/gamemodes/dynamic/dynamic_rulesets.dm

```diff
@@ -0,0 +1,209 @@
+/datum/dynamic_ruleset
+	var/name = "" // For admin logging
+	var/ruletype = "" // For admin logging, do not change unless making a new rule type.
+	var/persistent = FALSE // If set to TRUE, the rule won't be discarded after being executed, and dynamic will call rule_process() every SSTicker tick
+	var/repeatable = FALSE // If set to TRUE, dynamic mode will be able to draft this ruleset again later on. (doesn't apply for roundstart rules)
+	var/repeatable_weight_decrease = 2 // If set higher than 0, decreases weight by itself causing the ruleset to appear less often the more it is repeated.
+	var/list/mob/candidates = list() // List of players that are being drafted for this rule
+	var/list/datum/mind/assigned = list() // List of players that were selected for this rule
+	var/antag_flag = null // Preferences flag such as ROLE_WIZARD that need to be turned on for players to be antag
+	var/datum/antagonist/antag_datum = null
+	var/list/protected_roles = list() // If set, and config flag protect_roles_from_antagonist is false, then the rule will pick players from these roles.
+	var/list/restricted_roles = list() // If set, rule will deny candidates from those roles
+	var/list/exclusive_roles = list() // If set, rule will only accept candidates from those roles, IMPORTANT: DOES NOT WORK ON ROUNDSTART RULESETS.
+	var/list/enemy_roles = list() // If set, there needs to be a certain amount of players doing those roles (among the players who won't be drafted) for the rule to be drafted IMPORTANT: DOES NOT WORK ON ROUNDSTART RULESETS.
+	var/required_enemies = list(1,1,0,0,0,0,0,0,0,0) // If enemy_roles was set, this is the amount of enemy job workers needed per threat_level range (0-10,10-20,etc) IMPORTANT: DOES NOT WORK ON ROUNDSTART RULESETS.
+	var/required_candidates = 0 // The rule needs this many candidates (post-trimming) to be executed (example: Cult need 4 players at round start)
+	var/weight = 5 // 1 -> 9, probability for this rule to be picked against other rules
+	var/cost = 0 // Threat cost for this rule.
+
+	var/flags = 0
+
+	// For midround polling
```

Comment:
// Stores candidates that hit yes on a midround prompt
Is candidates the right word here?

Reply:
I will move this to midround rulesets and I got a comment locally for this already.

Reply:
I see, that certainly makes sense!